### PR TITLE
Fixed types

### PR DESCRIPTION
**Pull request by EvanCarroll**
Closes out #48 brings rxjs-websockets in line with @types/ws

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 export interface IWebSocket {
-  close()
-  send(data: string | ArrayBuffer | Blob)
-  onopen?: (event: Event) => any
-  onclose?: (event: CloseEvent) => any
-  onmessage?: (event: MessageEvent) => any
-  onerror?: (event: ErrorEvent) => any
+  close(): void
+  send(data: WebSocket.Data)
+  onopen?: (event: { target: WebSocket }) => void;
```
Comment (EvanCarroll) [inline src/index.ts]: Taken from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/ws

**Comment by ohjames** [comment]
This breaks the typings for use in the browser. ws types should be updated.

File: src/index.ts (hunk 1 above)
Comment (ohjames) [reply src/index.ts]: The `send` typings here are better though, I need to integrate that part of the change.

**Comment by ohjames** [comment]
Hm I'll think about this.

File: src/index.ts [hunk 2]
```diff
@@ -1,17 +1,18 @@
 import { Observable, Subscription, BehaviorSubject } from 'rxjs'
+import * as WebSocket from 'ws';
```
Comment (ohjames) [inline src/index.ts]: Can't include a Dev dependency in a source file.

File: src/index.ts [hunk 3]
```diff
@@ -49,16 +50,16 @@ export default function connect(
...
     }
 
-    socket.onerror = (error: ErrorEvent) => {
+    socket.onerror = (error) => {
       closed()
       observer.error(error)
     }
 
-    socket.onclose = (event: CloseEvent) => {
+    socket.onclose = (event) => {
```
Comment (ohjames) [inline src/index.ts]: This will mess up "implicit any" checks people have defined.

File: src/index.ts (hunk 2 above)
Comment (ohjames) [reply src/index.ts]: I guess it's okay here because `@types/ws` is a dependency.

**Comment by ohjames** [comment]
I tried out this version in a client project also and it breaks it... need to write something that supports both environments.

**Comment by ohjames** [comment]
@EvanCarroll I'm changing the interfaces now to only specify the types in the interfaces that are actually used by rxjs-websockets. This will allow it to be used with a much wider variety of websocket implementations. Thanks so much for opening this PR, sorry for being too dense to immediately realise how important it was. I'll update the linked issue when the code is ready. Thanks again!

**Pull request closed without merging**